src/datasets/arrow_dataset.py

```diff
@@ -537,16 +550,16 @@ class DatasetTransformationNotAllowedError(Exception):
     pass
 
 
-def transmit_format(func):
+def transmit_format(func: Callable) -> Callable:
     """Wrapper for dataset transforms that recreate a new Dataset to transmit the format of the original dataset to the new dataset"""
 
     @wraps(func)
-    def wrapper(*args, **kwargs):
+    def wrapper(*args: Any, **kwargs: Any) -> Union[dict, Dataset, DatasetDict]:
```

Comment:
```suggestion
    def wrapper(*args: Any, **kwargs: Any) -> Union[Dataset, DatasetDict]:
```

Reply:
This suggestion introduces the following error on the return type

```
Type "dict[Unknown, Unknown] | DatasetDict | Unknown | Dataset" is not assignable to return type "Dataset | DatasetDict"
  Type "dict[Unknown, Unknown] | DatasetDict | Unknown | Dataset" is not assignable to type "Dataset | DatasetDict"
    Type "dict[Unknown, Unknown]" is not assignable to type "Dataset | DatasetDict"
      "dict[Unknown, Unknown]" is not assignable to "Dataset"
      "dict[Unknown, Unknown]" is not assignable to "DatasetDict"PylancereportReturnType
(variable) out: dict[Unknown, Unknown] | DatasetDict | Unknown | Dataset
```